test/python/circuit/library/test_phase_oracle.py

```diff
@@ -37,7 +37,11 @@ class TestPhaseOracle(QiskitTestCase):
     @unpack
     def test_evaluate_bitstring(self, expression, input_bitstring, expected):
         """PhaseOracle(...).evaluate_bitstring"""
-        oracle = PhaseOracle(expression)
+        with self.assertWarnsRegex(
+            DeprecationWarning,
+            expected_regex="BooleanExpression`` is deprecated as of qiskit 1.4",
+        ):
+            oracle = PhaseOracle(expression)
```

Comment:
The `PhaseOracle` is not getting deprecated, it should not emit a deprecation warning. Instead we can filter the warning in the `PhaseOracle` code in 1.4 until we get the tweedledum-free version in 2.0.

Reply:
Done in 4510aee